### ISSUE TITLE
Color Detection Critical Bug in predict.webcam() function 

### ISSUE THREAD

**Issue by AmirKaseb** [Deci-AI/super-gradients#1693]
### 🐛 Describe the bug

# Bug Overview
I have found a critical bug in the predict.webcam() function , the function should capture your default device to give you a real time model prediction , it works fine for object detection , but there was a crucial error we find when we trying to train model on color detection  . 

# Before Fixing Bug
![Screenshot (2148)](https://github.com/Deci-AI/super-gradients/assets/109258480/97924a50-92e0-4ebd-8c6e-44a73f04ff9f)
As you see it detects blue ball as red , We were pretty sure about our labeling and our dataset  , things get more weird when it starts to detect red and orange as blue balls .

The issue was in predict_webcam() function who first treat the picture in  bgr mode then converting it after applying the model which would be non-effective way for color detection .

After researches we have fixed the issue , by editing 
# Process of Fixing the  Bug
**_C:\Users\moham\Downloads\super-gradients-master.zip\super-gradients-master\src\super_gradients\training\utils\media_** 

```python
    def _display_single_frame(self) -> None:
        """Read a single frame from the video capture device, apply any specified frame processing,
        and display the resulting frame in the window.

        Also updates the FPS counter and displays it in the frame.
        """
        _ret, frame = self.cap.read()
         frame= cv2.cvtColor(frame,cv2.COLOR_BGR2RGB) //Making the processing on RGB Photo to prevent confliction on color detection 
         
        if self.frame_processing_fn:
            frame = self.frame_processing_fn(frame)

        _write_fps_to_frame(frame, self.fps)
        frame= cv2.cvtColor(frame,cv2.COLOR_RGB2BGR) //then reversing the operation to optain an rgb output
        cv2.imshow(self.window_name, frame)
``` 

# The Result :- 

![image](https://github.com/Deci-AI/super-gradients/assets/109258480/7a3517c2-d89a-47bb-bc82-7b4eedbc82bd)

The Color Detection work fine on real time capturing without any errors , So i wish you fix this issue for further people who would use the model in color detection modules using real life camera .

### Versions

Collecting environment information...
PyTorch version: 2.1.1+cu118
Is debug build: False
CUDA used to build PyTorch: 11.8
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Enterprise
GCC version: (x86_64-posix-seh, Built by strawberryperl.com project) 8.3.0
Clang version: Could not collect
CMake version: version 3.25.1
Libc version: N/A

Python version: 3.10.11 (tags/v3.10.11:7d4cc5a, Apr 5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.19045-SP0
Is CUDA available: True
CUDA runtime version: 11.8.89
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: Quadro P2000
Nvidia driver version: 537.42
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.8\bin\cudnn_ops_train64_8.dll
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture=9
CurrentClockSpeed=2592
DeviceID=CPU0
Family=198
L2CacheSize=1536
L2CacheSpeed=
Manufacturer=GenuineIntel
MaxClockSpeed=2592
Name=Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
ProcessorType=3
Revision=

Versions of relevant libraries:
[pip3] numpy==1.23.0
[pip3] onnx==1.13.0
[pip3] onnx-graphsurgeon==0.3.27
[pip3] onnx-simplifier==0.4.35
[pip3] onnxruntime==1.13.1
[pip3] torch==2.1.1+cu118
[pip3] torchaudio==2.1.1+cu118
[pip3] torchmetrics==0.8.0
[pip3] torchvision==0.16.1+cu118
[conda] Could not collect

**Comment by BloodAxe** [comment]
Thanks for spotting this. Are you willing to open a PR with the fix?

**Comment by AmirKaseb** [comment]
yes sure can you please refer me with instructions to pull request

**Comment by BloodAxe** [comment]
If you follow [contributing guide](https://github.com/Deci-AI/super-gradients/blob/master/CONTRIBUTING.md) you should be good. In the nutshell - have code formatting done right and use signed commits.
